### PR TITLE
Sync test  of Temporal.Calendar.p*.fields to 1750

### PR DESCRIPTION
**Pull request by FrankYFTang**
https://github.com/tc39/proposal-temporal/pull

### CODE REVIEW

**Comment by FrankYFTang** [comment]
@ptomato @Ms2ger @cjtenny @jugglinmike @ryzokuken

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js [hunk 1]
```diff
@@ -10,22 +10,34 @@ info: |
...
       while (i++ < 1000001) {
         yield "garbage " + i;
       }
   }
 }
-assert(
-  compareArray(cal.fields(fields), Array.from(fields)),
-  'compareArray(cal.fields(fields), Array.from(fields)) must return true'
-);
+assert.throws( RangeError, () => cal.fields(fields), "Garbage content");
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/long-input.js:41]:
```suggestion
assert.throws(RangeError, () => cal.fields(fields), "Garbage content");
```

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js [hunk 2]
```diff
@@ -10,22 +10,34 @@ info: |
...
+// stop after the third one.
+assert.sameValue(3, i);
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/long-input.js:43]:
```suggestion
assert.sameValue(i, 3);
```

File: test/built-ins/Temporal/Calendar/prototype/fields/repeated-throw.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+let cal = new Temporal.Calendar("iso8601")
+let i = 0;
+const fields = {
+  *[Symbol.iterator]() {
+      yield "month";
+      i++
+      yield "year";
+      i++
+      yield "year";
+      i++
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/repeated-throw.js:34]:
```suggestion
      i++;
      yield "year";
      i++;
      yield "year";
      i++;
```

File: test/built-ins/Temporal/Calendar/prototype/fields/reverse.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     yield "second";
+     yield "minute";
+     yield "hour";
+     yield "day";
+     yield "monthCode";
+     yield "month";
+     yield "year";
+  }
+}
+assert(compareArray(cal.fields(fields), Array.from(fields)),
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/reverse.js:41]:
```suggestion
assert.compareArray(cal.fields(fields), Array.from(fields),
```

File: test/built-ins/Temporal/Calendar/prototype/fields/reverse.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    'valid fields should return true even if they are in reversed order of the spec');
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/reverse.js:42]: I found this sentence a bit hard to follow, so maybe:
```suggestion
    'valid fields should be supported even if they are in reversed order of the spec');
```

**Review by Ms2ger — DISMISSED** [review]

**Comment by rwaldron** [comment]
This cannot be merged until there is affirmatives from @ptomato @Ms2ger and @jugglinmike

File: test/built-ins/Temporal/Calendar/prototype/fields/reverse.js (hunk 4 above)
Comment (rwaldron) [reply test/built-ins/Temporal/Calendar/prototype/fields/reverse.js:41]: Yes, and every where else as well.

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js [hunk 6]
```diff
@@ -10,22 +10,34 @@ info: |
...
-      let i = 0;
+      // The first three are valid values
+      yield "year";
+      i++;
+      yield "month";
+      i++;
+      yield "monthCode";
+      // The third one are wrong
       while (i++ < 1000001) {
         yield "garbage " + i;
```
Comment (jugglinmike) [inline test/built-ins/Temporal/Calendar/prototype/fields/long-input.js:37]: The loop is misleading, now. Would you mind simplifying it to `i++; yield "garbage"; i++;`?

**Comment by FrankYFTang** [comment]
PTAL

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js [hunk 7]
```diff
@@ -10,22 +10,33 @@ info: |
...
-        yield "garbage " + i;
-      }
+      // The first three are valid values
+      yield "year";
+      i++;
+      yield "month";
+      i++;
+      yield "monthCode";
+      i++;
+      // The third one are wrong
```
Comment (Ms2ger) [inline test/built-ins/Temporal/Calendar/prototype/fields/long-input.js:36]:
```suggestion
      // The fourth one is wrong
```

**Review by Ms2ger — APPROVED** [review]

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js (hunk 7 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Calendar/prototype/fields/long-input.js]: ha ha, sure....

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js [hunk 8]
```diff
@@ -10,22 +10,33 @@ info: |
   ## 12.4.21 Temporal.Calendar.prototype.fields ( fields )
```
Comment (ptomato) [inline test/built-ins/Temporal/Calendar/prototype/fields/long-input.js:10]: (in lines 6–8) description is now out of date, should be updated to something like "Temporal.Calendar.prototype.fields will throw when its input iterable yields an invalid field"

File: test/built-ins/Temporal/Calendar/prototype/fields/repeated-throw.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.fields
+description: >
+  Temporal.Calendar.prototype.fields will take iterable of any size and any string
```
Comment (ptomato) [inline test/built-ins/Temporal/Calendar/prototype/fields/repeated-throw.js:7]: Maybe something like "Temporal.Calendar.prototype.fields will throw if its input iterable yields the same value twice"?

File: test/built-ins/Temporal/Calendar/prototype/fields/reverse.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.fields
+description: >
+  Temporal.Calendar.prototype.fields will take iterable of any size and any string
```
Comment (ptomato) [inline test/built-ins/Temporal/Calendar/prototype/fields/reverse.js:7]: Same here, needs a new description.

**Review by ptomato — APPROVED** [review]
LGTM, except for the descriptions.

**Comment by FrankYFTang** [comment]
@ptomato PTAL

**Review by ptomato — APPROVED** [review]

File: test/built-ins/Temporal/Calendar/prototype/fields/long-input.js (hunk 6 above)
Comment (jugglinmike) [reply test/built-ins/Temporal/Calendar/prototype/fields/long-input.js]: This request isn't quite resolved--the second `i++` is important to ensure that implementations don't first exhaust the generator and then validate the emitted value.
